src/libstd/thread/mod.rs

```diff
@@ -787,12 +787,16 @@ pub fn park_timeout_ms(ms: u32) {
 ///
 /// let timeout = Duration::from_secs(2);
 /// let beginning_park = Instant::now();
-/// park_timeout(timeout);
 ///
-/// while beginning_park.elapsed() < timeout {
-///     println!("restarting park_timeout after {:?}", beginning_park.elapsed());
-///     let timeout = timeout - beginning_park.elapsed();
-///     park_timeout(timeout);
+/// let mut timeout_remaining = timeout;
+/// loop {
+///     park_timeout(timeout_remaining);
+///     let elapsed = beginning_park.elapsed();
+///     if elapsed >= timeout {
```

Comment:
Nah, I think it is right as-is.

In the loop, `elapsed` is the total duration since creation of `beginning_park` (created outside of the loop). It is a rough measure of the total duration we've parked for thus far.
`timeout` is the total desired park duration.
Thus, comparing the two seems okay (we're comparing two "totals").
If we need to park some more, we calculate the difference between the total desired park duration and the total elapsed park duration, and use that value in yet-another-park.